oauth2_http/javatests/com/google/auth/oauth2/StsRequestHandlerTest.java

```diff
@@ -0,0 +1,319 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.testing.http.MockLowLevelHttpRequest;
+import com.google.api.client.util.GenericData;
+import com.google.api.client.util.Joiner;
+import com.google.auth.TestUtils;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.auth.oauth2.StsTokenExchangeRequest.ActingParty;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.Test;
+import org.junit.function.ThrowingRunnable;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link StsRequestHandler}. */
+@RunWith(JUnit4.class)
+public final class StsRequestHandlerTest {
+  private static final String TOKEN_EXCHANGE_GRANT_TYPE =
+      "urn:ietf:params:oauth:grant-type:token-exchange";
+  private static final String CLOUD_PLATFORM_SCOPE =
+      "https://www.googleapis.com/auth/cloud-platform";
+  private static final String DEFAULT_REQUESTED_TOKEN_TYPE =
+      "urn:ietf:params:oauth:token-type:access_token";
+  private static final String TOKEN_URL = "https://www.sts.google.com";
+  private static final String CREDENTIAL = "credential";
+  private static final String SUBJECT_TOKEN_TYPE = "subjectTokenType";
+
+  // Optional params.
+  private static final String AUDIENCE = "audience";
+  private static final String RESOURCE = "resource";
+  private static final String ACTOR_TOKEN = "actorToken";
+  private static final String ACTOR_TOKEN_TYPE = "actorTokenType";
+  private static final String REQUESTED_TOKEN_TYPE = "requestedTokenType";
+  private static final String INTERNAL_OPTIONS = "internalOptions";
+  private static final String REFRESH_TOKEN = "refreshToken";
+  private static final List<String> SCOPES = Arrays.asList("scope1", "scope2", "scope3");
+
+  // Headers.
+  private static final String CONTENT_TYPE_KEY = "content-type";
+  private static final String CONTENT_TYPE = "application/x-www-form-urlencoded";
+  private static final String ACCEPT_ENCODING_KEY = "accept-encoding";
+  private static final String ACCEPT_ENCODING = "gzip";
+  private static final String CUSTOM_HEADER_KEY = "custom_header_key";
+  private static final String CUSTOM_HEADER_VALUE = "custom_header_value";
+
+  private static final String INVALID_REQUEST = "invalid_request";
+  private static final String ERROR_DESCRIPTION = "errorDescription";
+  private static final String ERROR_URI = "errorUri";
+
+  private static final MockStsServiceTransportFactory MOCK_HTTP_TRANSPORT_FACTORY =
```

Comment:
I don't see anything needing a TransportFactory as an argument - should we just instantiate the Transport in each test?

Reply:
We can just hold the transport directly - fixed.